### PR TITLE
Drag and Drop Component Improvement

### PR DESCRIPTION
**Pull request by BarryPaneer**
# Task
 - https://innso.atlassian.net/browse/TRIB-143

# Related PRs:
 - https://github.com/Learningtribes/platform/pull/1695
 - https://github.com/Learningtribes/studio-frontend/pull/3
 - https://github.com/Learningtribes/triboo-theme/pull/660

 # Doc
 - [FIGMA Design](https://www.figma.com/file/zfEb9v2zgSEP5i3yu74MPe/Triboo-Improvement-Drag-%26-Drop-component?node-id=918%3A3717&t=HNMKH200Inbl0Ded-1)
 - [testing site](https://studio-preprod.learning-tribes.com/)

# Feedbacks
 - [Quentin LUONG VAN ::: V_1](https://sitel-my.sharepoint.com/:p:/p/barry_bao/EYHE5CVuSUZInZQGwcmH38gBIGTdJC2c29Q4osHGd_oudQ?e=jveix3)
 - [Philippe Zeganath ::: V_1](https://sitel-my.sharepoint.com/:p:/p/barry_bao/EbCczprV6ApMgBudnqNcVyABo0Ommm7FY4o1Ywth3RioJw?e=abp4eN)
 - [Philippe Zeganath ::: V_2](https://sitel-my.sharepoint.com/:w:/p/barry_bao/EUWmkefKqC5KhiQM-CtPSTEB2D09cy0hPinh1miHiz3i2A?e=XZWIQs)
 - [Issue Course in preprod](https://studio-preprod.learning-tribes.com/container/block-v1:griky+CNN001+2020_T1+type@vertical+block@f912b2daee664bdc823f425bc154a753?action=new) ( `Qunentin LUONG VAN` )

# Demo 
 - [Testing on preprod with old version](https://studio-preprod.learning-tribes.com/course/course-v1:griky+bcs_1024000+2022-10-10)
    - https://preprod.learning-tribes.com/courses/course-v1:griky+bcs_1024000+2022-10-10/courseware/f722d2176ce0488cb0fb719c420aede2/bc1f9286f6b7493ab341fbf93571a52d/
 - [Testing on preprod with new version](https://studio-preprod.learning-tribes.com/course/course-v1:griky+bcs_1024001+2022-10-10)
    - https://preprod.learning-tribes.com/courses/course-v1:griky+bcs_1024001+2022-10-10/courseware?activate_block_id=block-v1%3Agriky%2Bbcs_1024001%2B2022-10-10%2Btype%40course%2Bblock%40course

# Translations
 - [Online Document](https://sitel-my.sharepoint.com/:x:/r/personal/barry_bao_sitel_com/_layouts/15/Doc.aspx?sourcedoc=%7B4B4C414E-4856-4332-A882-12B89D94D081%7D&file=Drag%20%26%20Drop%20XBlock%20related%20translations.xlsx&action=default&mobileredirect=true&DefaultItemOpen=1&login_hint=Barry.Bao%40sitel.com&ct=1685668668418&wdOrigin=OFFICECOM-WEB.MAIN.REC&cid=c07aa94a-a383-44cc-9837-d9f5e46c2543&wdPreviousSessionSrc=HarmonyWeb&wdPreviousSession=9a705e79-85bc-4bed-9537-205e049db7b7)

 # APIs:
 - [Red Save Button] : [POST] `https://.../xblock/block-v1:.../handler/studio_submit` 

# How to install an old version
 - `pip install git+https://github.com/Learningtribes/xblock-drag-and-drop-v2.git@117450f158953a904afe8eb78731d7041152939e`

# Tests 
 - https://studio-demo.triboolearning.com/container/block-v1:demo+BIO101+T1_2023+type@vertical+block@aff955eb679c4268b34cfb41f8063012
 - https://studio-demo.triboolearning.com/container/block-v1:demo+Stonedemocourse+2022Q1+type@vertical+block@acd6c6aa1bf64d24b7b2f308b36784e3
 - Tabs
image

image

image

image

image

image

image

image

image

image

image
( Opened in Mobile )
image
( `Reached` maximum attempts )
image
( `Clicked "Show Answer" Button )
image

### CODE REVIEW

File: drag_and_drop_v2/drag_and_drop_v2.py [hunk 1]
```diff
@@ -63,7 +75,7 @@ class DragAndDropBlock(
         display_name=_("Title"),
         help=_("The title of the drag and drop problem. The title is displayed to learners."),
         scope=Scope.settings,
-        default=_("Drag and Drop"),
+        default=_("Drag & Drop"),
```
Comment (noahzaozao) [inline drag_and_drop_v2/drag_and_drop_v2.py]: in my devstack, the "&" cause the "Drag and Drop" button not show up in the "Add New Compontent" -> "Quiz" list

Comment (BarryPaneer) [reply drag_and_drop_v2/drag_and_drop_v2.py]: image

 - Pls git pull latest source code of `platform`. This issue has been fixed.

**Comment by BarryPaneer** [comment]
- Fixed Style of "Add Zone Button" : 
image

 - When mouse hovering on the button : 
image

**Comment by BarryPaneer** [comment]
- Fixed style of "Show Answer" : 
image

 - Hovering on it : 
image

**Comment by BarryPaneer** [comment]
- Fixed : predefined colors of zones in Answer Cards ( `Zone 1` / `Zone 2` / `Answer 1` are all new added objects ) : 
image

**Comment by BarryPaneer** [comment]
image
 ( 2 / 9 )

**Comment by BarryPaneer** [comment]
image
image
 ( 7 / 9 )

**Comment by BarryPaneer** [comment]
image
image
( `Editor View` )
image
(`Student View`)

**Comment by BarryPaneer** [comment]
👉  Hide `Show Answer` Button if it's not available 
image
( Not reaching maximum attempts )
image
( Reached maximum attempts )
image
( Clicked Show Answer Button )

**Review by lpfaender — APPROVED** [review]

**Pull request merged**